src/System.Memory/tests/Performance/Perf.ReadOnlySequence.Slice.cs

```diff
@@ -0,0 +1,1069 @@
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                int localInt = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        ReadOnlySequence<byte> sliced = buffer;
+                        long sliceLen = sliced.Length;
+                        while (sliceLen > 0)
+                        {
+                            sliceLen -= offset;
+                            SequencePosition end = sliced.GetPosition(sliceLen);
+                            SequencePosition start = sliced.GetPosition(offset);
+                            sliceLen -= offset;
+                            sliced = sliced.Slice(start, end);
+                            localInt ^= sliced.Start.GetInteger();
+                        }
+                    }
+                }
+                _volatileInt = localInt;
+            }
+        }
+
+        #endregion
+
+        #region Byte_Empty
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        private static void Byte_Empty()
+        {
+            ReadOnlySequence<byte> buffer = ReadOnlySequence<byte>.Empty;
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                int localInt = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        ReadOnlySequence<byte> sliced = buffer.Slice(0L);
+                        localInt ^= sliced.Start.GetInteger();
+
+                        sliced = sliced.Slice(0L, 0L);
+                        localInt ^= sliced.Start.GetInteger();
+
+                        sliced = sliced.Slice(0L, sliced.End);
+                        localInt ^= sliced.Start.GetInteger();
+
+                        sliced = sliced.Slice(sliced.Start);
+                        localInt ^= sliced.Start.GetInteger();
+
+                        sliced = sliced.Slice(sliced.Start, 0L);
+                        localInt ^= sliced.Start.GetInteger();
+
+                        sliced = sliced.Slice(sliced.Start, sliced.End);
+                        localInt ^= sliced.Start.GetInteger();
+
+                        sliced = sliced.Slice(0, 0);
+                        localInt ^= sliced.Start.GetInteger();
+
+                        sliced = sliced.Slice(0, sliced.End);
+                        localInt ^= sliced.Start.GetInteger();
+
+                        sliced = sliced.Slice(sliced.Start, 0);
+                        localInt ^= sliced.Start.GetInteger();
+                    }
+                }
+                _volatileInt = localInt;
+            }
+        }
+
+        #endregion
+
+        #region Byte_Default
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        private static void Byte_Default()
```

Comment:
Same with this one. Remove the test please.

Reply:
I can split this test to each Slice method type.